### ISSUE TITLE
[Deno] error when doing bundle or compile

### ISSUE THREAD

**Issue by cinnamon-bun** [earthstar-project/stone-soup#125]
## What's the problem you want solved?

Deno has a known problem bundling `std/node`, resulting in 'Uncaught ReferenceError: Cannot access 'NodeTypeError' before initialization'

It emits the code in the wrong order resulting in this error:

Using our `esm-first` branch:
```sh
deno bundle --import-map=import_map.json --config=deno.json --no-check src/example-app.ts out.js
deno run -A out.js

error: Uncaught ReferenceError: Cannot access 'NodeTypeError' before initialization
class ERR_INVALID_ARG_TYPE extends NodeTypeError {
                                   ^
    at file:///Users/rabbit/projects/stone-soup-deno/out.js:8318:36
```
## Is there a solution you'd like to recommend?
We have to wait for Deno to fix it.

Deno has several issues tracking this problem.  It happens in all the latest versions of Deno including canary.

* https://github.com/denoland/deno/search?q=deno+Cannot+access+before+initialization&type=issues
* https://github.com/denoland/deno/issues/11286
* https://github.com/denoland/deno/issues/13061
* https://github.com/denoland/deno/issues/13170

**Comment by sgwilym** [comment]
As the bundle no longer includes any std/node code, `NodeTypeError` is no longer included. I'll add a CI step for this.

**Comment by cinnamon-bun** [comment]
This was fixed in `std@0.123.0`

https://github.com/denoland/deno/issues/13503
